packages/angular-material/src/other/master-detail/master.ts

```diff
@@ -169,22 +170,22 @@ export class MasterListComponent extends JsonFormsArrayControl {
     const { data, path, schema, uischema } = props;
     const controlElement = uischema as ControlElement;
     this.propsPath = props.path;
-    const detailUISchema =
-      controlElement.options.detail ||
-      findUISchema(
+    const detailUISchema = findUISchema(
         props.uischemas,
         schema,
         `${controlElement.scope}/items`,
         props.path,
-        'VerticalLayout'
+        'VerticalLayout',
+        controlElement
       );
 
     const masterItems = (data || []).map((d: any, index: number) => {
-      const labelRefInstancePath = removeSchemaKeywords(
+      const labelRefInstancePath = controlElement.options?.labelRef && removeSchemaKeywords(
         controlElement.options.labelRef
       );
+      const isPrimitive = d && typeof d !== 'object';
```

Comment:
this is false for `0` and empty strings `''`

Reply:
Will fix, JS and auto conversion of types ...